src/modules/esl-drag-to-scroll/core/esl-drag-to-scroll.mixin.ts

```diff
@@ -0,0 +1,141 @@
+  /** Determines the scrolling axis. Options are 'x', 'y', or 'both' (default) */
+  axis: 'x' | 'y' | 'both';
+  /** Class name to apply during dragging. Defaults to 'dragging' */
+  cls: string;
+  /** Min distance in pixels to activate dragging mode. Defaults to 10 */
+  tolerance: number;
+  /** Prevent dragging if text is selected or not. Defaults to true */
+  selection: boolean;
+}
+
+/**
+ * ESLDragToScrollMixin - mixin to enable drag-to-scroll functionality for any scrollable container element
+ * @author Anna Barmina, Alexey Stsefanovich (ala'n)
+ *
+ * Use example:
+ * ```
+ * <div class="esl-scrollable-content" esl-drag-to-scroll>
+ *   <!-- Content here -->
+ * </div>
+ * ```
+ */
+export class ESLDragToScrollMixin extends ESLMixinElement {
+  public static override is = 'esl-drag-to-scroll';
+
+  /** Default configuration object */
+  public static DEFAULT_CONFIG: ESLDragToScrollConfig = {
+    axis: 'both',
+    cls: 'dragging',
+    tolerance: 10,
+    selection: true
+  };
+
+  /** Raw configuration object, provided via the `esl-drag-to-scroll` attribute */
+  @jsonAttr({name: ESLDragToScrollMixin.is}) public rawConfig: Partial<ESLDragToScrollConfig>;
+
+  /** Initial pointer event when dragging starts */
+  private startEvent: PointerEvent;
+
+  private startScrollLeft: number = 0;
+  private startScrollTop: number = 0;
+
+  private _isDragging: boolean = false;
+
+  private set isDragging(value: boolean) {
+    this._isDragging = value;
+    this.$$cls(this.config.cls, value);
+  }
+
+  /** Flag indicating whether dragging is in progress */
+  public get isDragging(): boolean {
+    return this._isDragging;
+  }
+
+  /** Merged configuration object combining default and raw configuration */
+  @memoize()
+  public get config(): ESLDragToScrollConfig {
+    return {...ESLDragToScrollMixin.DEFAULT_CONFIG, ...this.rawConfig};
+  }
+
+  /** Flag indicating whether text is selected */
+  public get hasSelection(): boolean {
+    const selection = document.getSelection();
+    if (!selection || !this.config.selection) return false;
+    // Prevents draggable state if the text is selected
+    return !selection.isCollapsed && this.$host.contains(selection.anchorNode);
+  }
+
+  protected override attributeChangedCallback(name: string, oldValue: string, newValue: string): void {
+    memoize.clear(this, 'config');
+  }
+
+  /** @returns the offset of the pointer event relative to the start event */
+  public getEventOffset(event: PointerEvent): Point {
+    if (!this.startEvent || event.type === 'pointercancel') return {x: 0, y: 0};
+    const x = event.clientX - this.startEvent.clientX;
+    const y = event.clientY - this.startEvent.clientY;
+    return {x, y};
+  }
+
+  /** Scrolls the host element by the specified offset */
+  public scrollBy(offset: Point): void {
+    if (this.config.axis === 'x' || this.config.axis === 'both') {
+      this.$host.scrollLeft = this.startScrollLeft - offset.x;
```

Comment:
That's super-super cosmetic, but mathematically, offset should be added, not subtracted.

Just in case you're going to do additional cleanup, I'd replace it with +, while calculating offset as "start" - "current" in the method above.

PS: We definitely need truthful pointer utilities in the future to simplify all calculations.